test/e2e/specs/editor/various/patterns.spec.js

```diff
@@ -252,4 +252,49 @@ test.describe( 'Synced pattern', () => {
 			},
 		] );
 	} );
+
+	// Check for regressions of https://github.com/WordPress/gutenberg/issues/26421.
+	test( 'allows conversion back to blocks when the reusable block has unsaved edits', async ( {
+		page,
+		requestUtils,
+		editor,
+	} ) => {
+		const paragraphId = 'paragraph-id';
+		const { id } = await requestUtils.createBlock( {
+			title: 'Pattern',
+			content: `<!-- wp:paragraph {"metadata":{"id":"${ paragraphId }","bindings":{"content":{"source":{"name":"pattern_attributes"}}}}} -->
+<p>Editable</p>
+<!-- /wp:paragraph -->`,
+			status: 'publish',
+		} );
+
+		await editor.insertBlock( {
+			name: 'core/block',
+			attributes: { ref: id },
+		} );
+
+		// Make an edit to the pattern.
+		await editor.canvas
+			.getByRole( 'document', { name: 'Block: Paragraph' } )
+			.focus();
+		await page.keyboard.type( 'edited ' );
+
+		// Convert back to regular blocks.
+		await editor.selectBlocks(
+			editor.canvas.getByRole( 'document', { name: 'Block: Pattern' } )
+		);
+		await editor.showBlockToolbar();
```

Comment:
This should not be necessary; the `clickBlockOptionsMenuItem` will handle that.

Reply:
Yeah, I'm just including it for clarity. It's clearer for me this way that `clickBlockOptionsMenuItem` is interacting with the block toolbar, but not some other UI on the page. But yeah, I agree it's unnecessary and `clickBlockOptionsMenuItem` should be clear enough 😅.